shared_libs/transformer_base_lib/transformer_base_lib/app_config.py

```diff
@@ -44,36 +44,31 @@ def read_env_config() -> AppConfig:
 
 @dataclass
 class TransformerConfig(AppConfig):
-    MAX_BATCH_SIZE: int
+    MAX_BATCH_SIZE: int | None
 
     @classmethod
-    def from_env_and_config_file(cls, config_path: str | None) -> "TransformerConfig":
+    def from_env_and_config_file(cls, config_path: str) -> "TransformerConfig":
+        logger = logging.getLogger(__name__)
         app_config = AppConfig.read_env_config()
 
         config = configparser.ConfigParser()
-        # MAX_BATCH_SIZE defaults to 1 when not configured to stop the reconnection loop
-        MAX_BATCH_SIZE = 1
-
-        if config_path:
-            logger.debug(
-                "Loading transformer configuration from %s (exists=%s)",
-                config_path,
-                os.path.exists(config_path),
-            )
+        config.read(config_path)
```

Comment:
assumes the path provided is valid - will error if not the case without setting a default